src/data/move.ts

```diff
@@ -2875,6 +2875,164 @@ export class StatStageChangeAttr extends MoveEffectAttr {
+      this.effectChanceOverride = 100; // effectChanceOverride used in the application of the actual secondary effect
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Used to apply the secondary effect to the target Pokemon
+   * @param user
+   * @param target
+   * @param move
+   * @param args
+   * @returns `true` if a secondary effect is successfully applied | 'false' if not
+   */
+  override apply(user: Pokemon, target: Pokemon, move: Move, args?: any[] | undefined): boolean | Promise<boolean> {
+    if (!super.apply(user, target, move, args)) {
+      return false;
+    }
+    let secondaryEffect: MoveEffectAttr;
+    const terrain = user.scene.arena.getTerrainType();
+    if (terrain !== TerrainType.NONE) {
+      secondaryEffect = this.determineTerrainEffect(terrain);
+    } else {
+      const biome = user.scene.arena.biomeType;
+      secondaryEffect = this.determineBiomeEffect(biome);
+    }
+    return secondaryEffect!.apply(user, target, move, []);
+  }
+
+  /**
+   * Determines the secondary effect based on terrain
+   * Takes precedence over biome
+   * Electric Terrain | Paralysis
+   * Misty Terrain    | SpAtk -1
+   * Grassy Terrain   | Sleep
+   * Psychic Terrain  | Speed -1
+   * @param terrain {@linkcode TerrainType}the current terrain
+   * @returns the chosen secondary effect {@linkcode MoveEffectAttr}
+   */
+  private determineTerrainEffect(terrain: TerrainType): MoveEffectAttr {
+    let secondaryEffect: MoveEffectAttr;
+    switch (terrain) {
+    case TerrainType.ELECTRIC:
+      secondaryEffect = new StatusEffectAttr(StatusEffect.PARALYSIS, false);
+      break;
+    case TerrainType.MISTY:
+      secondaryEffect = new StatStageChangeAttr([ Stat.SPATK ], -1, false);
+      break;
+    case TerrainType.GRASSY:
+      secondaryEffect = new StatusEffectAttr(StatusEffect.SLEEP, false);
+      break;
+    case TerrainType.PSYCHIC:
+      secondaryEffect = new StatStageChangeAttr([ Stat.SPD ], -1, false);
+      break;
+    }
+    return secondaryEffect!;
+  }
+
+  /**
+   * Determines the secondary effect based on biome
+   * Town, Metropolis, Slum, Dojo, Laboratory, Power Plant + Default | Paralysis
+   * Plains, Grass, Tall Grass, Forest, Jungle, Meadow               | Sleep
+   * Swamp, Mountain, Temple, Ruins                                  | Speed -1
+   * Ice Cave, Snowy Forest                                          | Freeze
+   * Volcano                                                         | Burn
+   * Fairy Cave                                                      | SpAtk -1
+   * Desert, Construction Site, Beach, Island, Badlands              | Accuracy -1
+   * Sea, Lake, Seabed                                               | Atk -1
+   * Cave, Wasteland, Graveyard, Abyss, Space                        | Flinch
+   * End                                                             | Def -1
+   * @param biome {@linkcode Biome} the current biome the battle is set in
```

Comment:
```suggestion
   * ```
   * Town, Metropolis, Slum, Dojo, Laboratory, Power Plant + Default | Paralysis
   * Plains, Grass, Tall Grass, Forest, Jungle, Meadow               | Sleep
   * Swamp, Mountain, Temple, Ruins                                  | Speed -1
   * Ice Cave, Snowy Forest                                          | Freeze
   * Volcano                                                         | Burn
   * Fairy Cave                                                      | SpAtk -1
   * Desert, Construction Site, Beach, Island, Badlands              | Accuracy -1
   * Sea, Lake, Seabed                                               | Atk -1
   * Cave, Wasteland, Graveyard, Abyss, Space                        | Flinch
   * End                                                             | Def -1
   * ```
   * @param biome - The current {@linkcode Biome} the battle is set in
```